src/main/java/de/tum/in/www1/artemis/service/ParticipationService.java

```diff
@@ -178,13 +178,14 @@ public SolutionProgrammingExerciseParticipation save(SolutionProgrammingExercise
      * repository / build plan related stuff for programming exercises. In the case of modeling or text exercises, it also initializes and stores the corresponding submission.
      *
      * @param exercise the exercise which is started
-     * @param username the name of the user who starts the exercise
+     * @param user the user who starts the exercise
      * @return the participation connecting the given exercise and user
      */
-    public StudentParticipation startExercise(Exercise exercise, String username) {
+    public StudentParticipation startExercise(Exercise exercise, User user) {
         // common for all exercises
         // Check if participation already exists
-        Optional<StudentParticipation> optionalStudentParticipation = findOneByExerciseIdAndStudentLoginAnyState(exercise.getId(), username);
+        var username = user.getLogin();
```

Comment:
This is not needed and can be removed.  Otherwise, use it in line `188`.

Reply:
I removed it